### PR TITLE
Port https://github.com/dotnet/aspnetcore/pull/38814 to 6.0

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by Pilchie** [comment]
The goal is to increase coverage in servicing?  Sounds good to me.

**Comment by pranavkm** [comment]
@dougbu had attempted to port this back to 6.0 yesterday: https://github.com/dotnet/aspnetcore/pull/38814#issuecomment-1032831047. I'm not super sure of the background on this though.

**Review by dougbu — COMMENTED** [review]
We need the following in this PR or the branch is going to get _more_ flaky
https://github.com/dotnet/aspnetcore/blob/main/eng/test-configuration.json#L11

That was visible in #38814 but not 9c29067c4087 for some reason

File: src/Middleware/Diagnostics/test/FunctionalTests/Diagnostics.FunctionalTests.csproj
Comment (dougbu) [inline src/Middleware/Diagnostics/test/FunctionalTests/Diagnostics.FunctionalTests.csproj:10]: > @dougbu had attempted to port this back to 6.0 yesterday: https://github.com/dotnet/aspnetcore/pull/38814#issuecomment-1032831047. I'm not super sure of the background on this though.

The background is we need this addition in release/6.0 to get PRs passing because LocalDB isn't reliable where we need it to be reliable.

The un-quarantining is also fine as long as the tests have been just as reliable in this branch. If the test or test infrastructure isn't up to the task, we should backport _only_ this change. I'm not familiar enough w/ the tests to know either way and am asking the gathered masses (as I would have if the backport bot could work)&hellip;

Comment (dougbu) [reply src/Middleware/Diagnostics/test/FunctionalTests/Diagnostics.FunctionalTests.csproj:10]: /btw this lack is also holding up #40073

Comment (dougbu) [reply src/Middleware/Diagnostics/test/FunctionalTests/Diagnostics.FunctionalTests.csproj:10]: > I'm not familiar enough w/ the tests to know either way and am asking the gathered masses (as I would have if the backport bot could work)…

Checked the analytics and the tests have been passing at 100% in this branch too 🎉

**Comment by wtgodbe** [comment]
> We need the following in this PR or the branch is going to get more flaky
https://github.com/dotnet/aspnetcore/blob/main/eng/test-configuration.json#L11

There is no `test-configuration.json` in this branch, is there more needed than just dropping it in?

**Comment by pranavkm** [comment]
@dougbu / @wtgodbe good to merge?

**Comment by dougbu** [comment]
> @dougbu / @wtgodbe good to merge?

Please add the missing eng/test-configuration.json file to allow un-quarantining in src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs. Fine to just copy the file from 'main' and un-quarantine the other test methods mentioned there later. Also fine to narrow things down to the one Helix-run retry.

**Review by wtgodbe — APPROVED** [review]
Looks good, but we should probably wait for next month at this point

**Review by dougbu — APPROVED** [review]

**Comment by dougbu** [comment]
Tests here are now failing because Blazor template tests are still running on Helix. Some differences between this PR and #38814 remain because src/ProjectTemplates/BlazorTemplates.Tests/BlazorTemplates.Tests.csproj doesn't appear to change here.

**Comment by wtgodbe** [comment]
@pranavkm please check with the CQBs before merging into release branches this far into the build window, changes were supposed to be in by EOD Monday

**Comment by pranavkm** [comment]
Sorry about that. I missed reading your previous comment and thought the approval meant we were good to get this in.

**Comment by dougbu** [comment]
Was the problem addressed in this PR blocking 6.0 dependency update PRs❔ Or, were the CQBs ignoring the problem (or problems) and merging on red❔

**Comment by wtgodbe** [comment]
Looks like the current 6.0 aspnetcore dependency PR has been red for a few days - I think CQBs aren't focused on that yet because of the ongoing ref pack issues in WindowsDesktop.

**Comment by dougbu** [comment]
Unfortunately things didn't turn green w/ this fix. The CG failures (which just turn the build orange fortunately) on Linux MUSL indicate we need either #39898 backported or another Arcade update (if @mmitche's `continueOnError` fix is now available for 6.0 branches). Should we take the time to do one or the other❔

**Comment by wtgodbe** [comment]
Let's backport https://github.com/dotnet/aspnetcore/pull/39898 for now

**Comment by dougbu** [comment]
Realized we have an Arcade build containing `continueOnError`. So, the orangeness is mostly aesthetic but I prefer we get to green if we have time. Thanks for starting the backport.

**Comment by wtgodbe** [comment]
https://github.com/dotnet/aspnetcore/pull/40158

**Pull request merged**
